quantum/process_keycode/process_autocorrect.c

```diff
@@ -0,0 +1,216 @@
+// Copyright 2021 Google LLC
+// Copyright 2021 @filterpaper
+// SPDX-License-Identifier: Apache-2.0
+// Original source: https://getreuer.info/posts/keyboards/autocorrection
+
+#include "process_autocorrect.h"
+#include <string.h>
+#include "keycode_config.h"
+
+#ifndef AUTOCORRECT_DATA_H
+#    define AUTOCORRECT_DATA_H "autocorrect_data.h"
+#endif
+
+#if __has_include(AUTOCORRECT_DATA_H)
+#    include AUTOCORRECT_DATA_H
+#else
+#    pragma message "Warning! Autocorrect is using the default library!"
+#    include "autocorrect_data_default.h"
+#endif
+
+/**
+ * @brief handler for determining if autocorrect should process keypress
+ *
+ * @param keycode Keycode registered by matrix press, per keymap
+ * @param record keyrecord_t structure
+ * @param typo_buffer_size passed along to allow resetting of autocorrect buffer
+ * @param mods allow processing of mod status
+ * @return true Allow autocorection
+ * @return false Stop processing and escape from autocorrect.
+ */
+__attribute__((weak)) bool process_autocorrect_user(uint16_t *keycode, keyrecord_t *record, uint8_t *typo_buffer_size, uint8_t *mods) {
+    switch (*keycode) {
+        case KC_LSFT:
+        case KC_RSFT:
+        case KC_CAPS:
+        case QK_TO ... QK_ONE_SHOT_LAYER_MAX:
+        case QK_LAYER_TAP_TOGGLE ... QK_LAYER_MOD_MAX:
+            // see quantum_keycodes.h for reference on these ranges
+            // QK_LAYER_TAP and QK_ONE_SHOT_MOD ranges handled below
+            return false;
+#ifndef NO_ACTION_TAPPING
+        case QK_MOD_TAP ... QK_MOD_TAP_MAX:
+#    ifndef NO_ACTION_LAYER
+        case QK_LAYER_TAP ... QK_LAYER_TAP_MAX:
+#    endif
+            if (!record->tap.count) {
+                return false; // Early return if tap-hold key is not being tapped.
+            }
+            *keycode &= 0xFF;
+            break;
+#endif
+#ifdef SWAP_HANDS_ENABLE
+        case QK_SWAP_HANDS ... QK_SWAP_HANDS_MAX:
+            if (*keycode >= 0x56F0 || !record->tap.count) {
+                return false;
+            }
+            *keycode &= 0xFF;
+            break;
+#endif
+#ifndef NO_ACTION_ONESHOT
+        case QK_ONE_SHOT_MOD ... QK_ONE_SHOT_MOD_MAX:
+            if ((*keycode & 0xF) != MOD_LSFT) {
+                typo_buffer_size = 0;
+            }
+            return false;
+            // fall through intentionally
+#endif
+        case QK_LSFT ... QK_LSFT + 255:
+        case QK_RSFT ... QK_RSFT + 255:
+            if (*keycode >= QK_LSFT && *keycode <= (QK_LSFT + 255)) {
+                *mods |= MOD_LSFT;
+            } else {
+                *mods |= MOD_LSFT;
+            }
+
```

Comment:
```suggestion
            if (*keycode >= QK_LSFT && *keycode <= (QK_LSFT + 255)) {
                *mods |= MOD_LSFT;
            } else {
                *mods |= MOD_RSFT;
            }
```
Is this the right intend for the right shift?